src/MicrosoftAuthentication.js

```diff
@@ -0,0 +1,36 @@
+const MicrosoftAuthFlow = require('./MicrosoftAuthFlow');
+
+/**
+ * Authenticates with Mincrosoft through user credentials, then
+ * with Xbox Live, Minecraft, checks entitlements and returns profile
+ *
+ * @function
+ * @param {object} client - The client passed to protocol
+ * @param {object} options - Client Options
+ */
+async function authenticatePassword(client, options) {
+    throw Error('Not implemented')
```

Comment:
Only difference for password based flow would be the ms auth part, the xbox auth and part would be the same. So this can be added to msauthflow or a new class which extends it

Reply:
I am thinking we should expose MsAuthFlow and just let the user set a username and password, but if password isnt set just straight go to devicecode.

Reply:
Yeah, that works. The options should be explicit also.